### ISSUE TITLE
explore secagg performance/optimization

### ISSUE THREAD

**Issue by ErwanDemairy** [fedbiomed/fedbiomed#798]

**Comment by rtaiello** [comment]
Hi @mvesin @srcansiz,

The idea I would like to implement is the following:
Now the node is first training the ML local model, and then encrypts it with https://github.com/fedbiomed/fedbiomed/blob/41eab3772a27dcb9935f34227ed9d1480fa75f33/fedbiomed/common/secagg/_jls.py#L611.
I attached a draw of what I would like to do:
![example drawio](https://github.com/fedbiomed/fedbiomed/assets/41542771/25f08b05-787a-48a5-8627-50765fdc6651)
So I would like to create a function like `pre_encrypt`
```
def pre_encrypt(tua:int, size_model:int) -> List[mpz]:
        cipher = []
        for counter in range(size_model):
            pre_cipher.append(self._pre_encrypt((counter << self.pp.bits // 2) | tau))
        return pre_cipher
    def _pre_encrypt(self, plaintext, tau):
        pre_ctxt = powmod(self.pp.H(tau), self.s, self.pp.nsquare)
        return pre_ctxt
```
And when the ML is trained we can compute `encrypt`
```
def encrypt(ptxt, pre_ctxt):
   final_ctxt = []
  for pt, p_ct in zip(ptxt, pre_ctxt):
           nude_ct = (self._public_param.n_modulus * pt + 1) % self._public_param.n_square
           final_ctxt.append((nude_ctxt * pt_ct)  % self._public_param.n_square)
   return final_ctxt
```
So basically I would like to know if is it possible to make this parallelization? If is yes where should it happen?

Thanks in advance!

**Comment by srcansiz** [comment]
Hi @rtaiello, 
I am closing this issue since you continue working on the issue #813 to increase the performance of secure aggregation.
